python/packages/nisar/workflows/gslc.py

```diff
@@ -198,8 +203,9 @@ def run(cfg):
                                           subswaths=sub_swaths)
 
                 # write geocoded blocks to respective HDF5 datasets
-                for gslc_dataset, gslc_data_blk in zip(gslc_datasets,
-                                                       gslc_data_blks):
+                for stats_complex, gslc_dataset, gslc_data_blk in zip(stats_complex_list,
+                                                                      gslc_datasets,
+                                                                      gslc_data_blks):
```

Comment:
It looks like this for loop iterates over the number of data blocks for each polarization, while the length of `stats_complex_list` corresponds to the number of polarizations (not the number of GSLC data blocks).

I think this change was introduced when you moved the statistics computation after the mantissa truncation, which is fine, but you probably need to adjust which element you’re pulling from `stats_complex_list`.